saleor/graphql/webhook/mutations.py

```diff
@@ -306,3 +316,80 @@ def perform_mutation(cls, _root, info: ResolveInfo, **data):
         manager = get_plugin_manager_promise(info.context).get()
         manager.event_delivery_retry(delivery)
         return EventDeliveryRetry(delivery=delivery)
+
+
+class WebhookDryRun(BaseMutation):
+    payload = JSONString(
+        description="JSON payload, that would be sent out to webhook's target URL."
+    )
+
+    class Arguments:
+        query = graphene.String(
+            description="The subscription query that defines the webhook event and its "
+            "payload.",
+            required=True,
+        )
+        object_id = graphene.ID(
+            description="The ID of an object to serialize.", required=True
+        )
+
+    class Meta:
+        description = "Performs a dry run of a webhook event."
+        permissions = (AppPermission.MANAGE_APPS,)
+        error_type_class = WebhookDryRunError
+
+    @classmethod
+    def validate_input(cls, info, **data):
+        query = data.get("query")
+        object_id = data.get("object_id")
+
+        event_type = get_event_type_from_subscription(query)
+        if not event_type:
+            raise_validation_error(
+                field="query",
+                message="Can't parse an event type from query.",
+                code=WebhookDryRunErrorCode.GRAPHQL_ERROR,
```

Comment:
Can we add specific errors for each type of failing validation? 
Returning a generic ones, doesn't allow you to build better error handling on the frontend part (as all or most of the errors returns the same error code). 
So here we should have something like: `WebhookDryRunErrorCode.UNABLE_TO_PARSE`.